archaius2-core/src/main/java/com/netflix/archaius/config/AbstractConfig.java

```diff
@@ -143,8 +142,7 @@ public String getString(String key) {
         if (value instanceof String) {
             return interpolator.create(getLookup()).resolve(value.toString());
         } else {
-            throw new UnsupportedOperationException(
-                    "Property values other than String not supported");
+            return value.toString();
```

Comment:
I'm a little wary of this. The config has explicit types getters. If we break the type check here, we lose the benefits of property types?